### ISSUE TITLE
docker network list returns very slow

### ISSUE THREAD

**Issue by xianlubird** [moby/moby#27729]
**Description**

**Steps to reproduce the issue:**
1. Run `docker network ls`

**Describe the results you received:**
I use etcd and overlay network in docker daemon. There're about 110 containers in my machine which has 2 core and 4G memory.
This command returns very slow, almost cost 5-7min.In some case, it won't return and hang forever

**Describe the results you expected:**
List network will list all containers and get endpoint of every container. It will get endpoint id from etcd and every connection with etcd will cost about 100ms. I think we should use `List` function to get all endpoint from etcd. `Libnetwork` should add this to its interface

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
Client:
 Version:      1.11.2.1
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   b5b6522
 Built:        Fri Jul 29 03:35:15 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.11.2.1
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   b5b6522
 Built:        Fri Jul 29 03:35:15 2016
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 107
 Running: 100
 Paused: 0
 Stopped: 7
Images: 8
Server Version: 1.11.2.1
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 252
 Dirperm1 Supported: false
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local nas ossfs acd
 Network: host bridge null overlay
Kernel Version: 3.13.0-32-generic
Operating System: Ubuntu 14.04.4 LTS
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 3.861 GiB
Name: c4b40254edf6549e295f0a9cafd0792f2-node16
ID: 6DRN:N2XT:6HEM:IANX:E2TD:PM44:SGIH:OGN7:3B2J:PT4M:AZTB:TTMZ
Docker Root Dir: /var/lib/docker
Debug mode (client): false
Debug mode (server): true
 File Descriptors: 76
 Goroutines: 165
 System Time: 2016-10-25T19:42:08.72092666+08:00
 EventsListeners: 6
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
Cluster store: etcd://111.111.111.111:2379/aiuwdhiauwgfiluawhfliuaw
Cluster advertise: 10.10.10.10:2376
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
#26812

**Comment by xianlubird** [comment]
#26812

**Comment by ccl0326** [comment]
any upgrade?

**Comment by xianlubird** [comment]
??????

**Comment by thaJeztah** [comment]
I think this should be addressed by https://github.com/moby/moby/pull/31710, which is included in Docker 17.04 and up; with that change, containers are no longer returned  by default, only if the `--verbose` option is used.
